.circleci/config.yml

```diff
@@ -45,6 +45,31 @@ jobs:
       - restore_cache:
           key: vendor-{{ checksum "Gopkg.lock" }}
       - run: go build -o machine-controller github.com/kubermatic/machine-controller/cmd/controller
+      - save_cache:
+          key: machine-controller-{{ .Revision }}
+          paths:
+            - machine-controller
+
+  publish:
+    machine: true
+    steps:
+      - run: >
+          sudo mkdir -p /go/src/github.com/kubermatic/machine-controller &&
+          sudo chown -R circleci:circleci /go/src/github.com/kubermatic/machine-controller
+      - restore_cache:
+          key: repo-{{ .Environment.CIRCLE_SHA1 }}
+      - restore_cache:
+          key: machine-controller-{{ .Revision }}
+      - restore_cache:
+          key: vendor-{{ checksum "/go/src/github.com/kubermatic/machine-controller/Gopkg.lock" }}
+      - run: >
+          cd /go/src/github.com/kubermatic/machine-controller &&
+          make docker-image-nodep &&
+          docker login --help &&
+          docker version &&
+          set -u &&
+          docker login -u "${DOCKERHUB_USER}" -p "${DOCKERHUB_PASS}" &&
+          make push-nodep
```

Comment:
why `&&` here?

Why not simply:
```
      - run: |
          make docker-image-nodep 
          ...
```

Reply:
Yeah, combined with a `set -x` that will do the same, I can change it